1-mandatory/2-people-i-know.js

```diff
@@ -393,15 +397,43 @@ You can see who people's colleagues are by seeing the "colleagues" array in each
 This time, I only want the full names ("<firstname> <lastname>") of my friends who are colleagues of hers.
 */
 
-let friendsWhoAreColleaguesOfStacie = [];
+let friendsWhoAreColleaguesOfStacie = []
+function Villarreal (arrayOfFriends) {
+  const result = [];
+  for (let i = 0; i < arrayOfFriends.length; i++) {
+    for (let j = 0; j < arrayOfFriends[i].colleagues.length; j++) {
+      if (arrayOfFriends[i].colleagues[j].name == "Stacie Villarreal") {
+        result.push(
+          `${arrayOfFriends[i].name.first} ${arrayOfFriends[i].name.last}`
+        );
+      }
+    }
+  }
+  return result;
+};
+
+friendsWhoAreColleaguesOfStacie = Villarreal(friends)
 /*
 5) Find "Multi-tasking" colleagues
 Next, I want you to find all of the colleagues of my friends who are good at "Multi-tasking"
 You can tell if they are good at "Multi-tasking" because they will have it listed in their skills
 This time, I only want the full names of the people who can multitask
 */
 
-let colleaguesWhoCanMultitask = [];
+let colleaguesWhoCanMultitask = []
+function multiTask (arrayOfFriends) {
+  const result = [];
+  for (let i = 0; i < arrayOfFriends.length; i++) {
+    for (let j = 0; j < arrayOfFriends[i].colleagues.length; j++) {
+      if (arrayOfFriends[i].colleagues[j].skills.includes("Multi-tasking")) {
+        result.push(arrayOfFriends[i].colleagues[j].name);
+      }
+    }
+  }
+  return result;
+};
```

Comment:
I have given a few examples with forEach and reduce, let me know if that helps. 

Also, if you are have to do multiple nestings, it's probably a good idea to break it down to multiple smaller functions